src/components/select/select-list/select-list.component.tsx

```diff
@@ -189,12 +189,33 @@ const SelectList = React.forwardRef(
       children,
     ]) as React.ReactElement<OptionProps>[];
 
+    function shallowEqual(
+      objA: Record<string, unknown>,
+      objB: Record<string, unknown>
+    ) {
+      const keysA = Object.keys(objA);
+
+      for (const key of keysA) {
+        if (objA[key] !== objB[key]) {
+          return false;
+        }
+      }
+      return true;
+    }
```

Comment:
**suggestion(non-blocking):** I _think_ you can do this with the `.every()` JS method. I tried it locally and it seemed to still work and all the tests (unit and PW) passed. I'm happy if you want to leave it the way you've done it as it works and it's easy to read. This is what I did though:

```
    function shallowEqual(
      objA: Record<string, unknown>,
      objB: Record<string, unknown>
    ): boolean {
      const keysA = Object.keys(objA);

      return keysA.every((key) => objA[key] === objB[key]);
    }
```

Reply:
you can short circuit it faster if you invert the logic and use `some`:

```
 function shallowEqual(
      objA: Record<string, unknown>,
      objB: Record<string, unknown>
    ): boolean {
      const keysA = Object.keys(objA);

      return !keysA.some((key) => objA[key] !== objB[key]);
    }